move_base/src/move_base.cpp

```diff
@@ -560,6 +566,86 @@ namespace move_base {
     return global_pose;
   }
 
+  //publish feedback based on latest_plan.Called in context of executeCycle.
+  void MoveBase::publishFeedback(const geometry_msgs::PoseStamped& current_position)
+  {
+    //calculate distance to goal from current position
+    std::pair<bool, double> distance_info = calculateDistanceToGoal(current_position);
+    if(!distance_info.first)
+    {
+      return;
+    }
+    //calculate average velocity
+    ros::Time current_time = ros::Time::now();
+    double dist = distance_info.second;
+    calculateAverageVelocity(dist, current_time);
+
+    //publish feedback after X(30) iterations
+    if (prev_feedback_info_.average_velocity_iteration > 30) {
+      move_base_msgs::Feedback msg;
+      msg.dist_to_goal = dist;
+      msg.time_to_goal = prev_feedback_info_.calculateTimeToGoal(dist);
+      feedback_distance_pub_.publish(msg);
+    }
+
+    //save previous feedback info
+    prev_feedback_info_.prev_time = current_time;
+    prev_feedback_info_.dist_to_goal = dist;
+  }
+
+  //calculate distance from current position to the goal
```

Comment:
Smth should be said that this is distance along the global plan. And better in the name too.